code/src/iris/src/lib/Runs.svelte

```diff
@@ -482,53 +485,111 @@
       // Reset for new search
       currentPage = 1;
       hasMoreResults = false;
-      totalLoadedResults = 0;
+      nextPageUrl = null;
+      allLoadedIds.clear();
     }
     error = null;
     
     try {
-      // Build query: use searchQuery from URL
-      let query = searchQuery.trim() || '';
-
-      // Request more than pageSize to detect if there are more results
-      const requestLimit = pageSize + 1;
-      const params: Record<string, unknown> = { 
-        tz: Intl.DateTimeFormat().resolvedOptions().timeZone,
-        limit: requestLimit,
-        offset: loadMore ? totalLoadedResults : 0
-      };
+      let response;
       
-      if (query) {
-        params.q = query;
+      if (loadMore && nextPageUrl) {
+        // Use the next page URL directly from the Link header
+        // Make direct fetch request to the URL provided by Link header
+        const fetchResponse = await fetch(nextPageUrl, {
+          method: 'GET',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          credentials: 'include',
+        });
+        
+        if (!fetchResponse.ok) {
+          throw new Error(`HTTP ${fetchResponse.status}: ${fetchResponse.statusText}`);
+        }
+        
+        const rawResponse = await fetchResponse.json();
+        
+        // Parse Link headers from the response
+        const linkHeader = fetchResponse.headers.get('Link') || fetchResponse.headers.get('link');
+        let linkHeaders: Record<string, string> | null = null;
+        if (linkHeader) {
+          // Simple parsing of Link header
+          linkHeaders = {};
+          const parts = linkHeader.split(/,\s*(?=<)/);
+          for (const part of parts) {
+            const match = part.match(/<([^>]+)>;\s*rel="([^"]+)"/);
+            if (match) {
+              linkHeaders[match[2]] = match[1];
+            }
+          }
+        }
+        
+        // Format response to match our expected structure
+        response = {
+          dirspaces: rawResponse.dirspaces || [],
+          hasMore: linkHeaders?.next !== undefined,
+          linkHeaders
+        };
+      } else {
+        // Build query for initial load
+        let query = searchQuery.trim() || '';
```

Comment:
What's this do?  if `searchQuery` is an empty string, then `searchQuery.trim()` is the same as `''`.